hydromt_wflow/components/config.py

```diff
@@ -1,43 +1,131 @@
-"""config component module."""
+"""Custom wflow config component module."""
 
-from typing import Any
+import logging
+from pathlib import Path
+from typing import Any, cast
 
 import tomlkit
 from hydromt.model import Model
-from hydromt.model.components.config import ConfigComponent
+from hydromt.model.components.base import ModelComponent
+from hydromt.model.steps import hydromt_step
+from tomli_w import dump as dump_toml
+from tomllib import load as load_toml
 
 from hydromt_wflow.utils import get_config, set_config
 
+__all__ = ["WflowConfigComponent"]
 
-class WflowConfigComponent(ConfigComponent):
-    """A class to manage the configuration of wflow models."""
+logger = logging.getLogger(f"hydromt.{__name__}")
+
+
+class WflowConfigComponent(ModelComponent):
+    """Manage the wflow configurations.
+
+    Parameters
+    ----------
+    model : Model
+        HydroMT model instance
+    filename : str
+        A path relative to the root where the configuration file will
+        be read and written if user does not provide a path themselves.
+        By default 'wflow_sbm.toml'
+    default_template_filename : str, optional
+        A path to a template file that will be used as default in the ``create``
+        method to initialize the configuration file if the user does not provide
+        their own template file. This can be used by model plugins to provide a
+        default configuration template. By default None.
+    """
 
     def __init__(
         self,
         model: Model,
         *,
         filename="wflow_sbm.toml",
-        default_template_filename: str | None = None,
     ):
-        super().__init__(
-            model=model,
-            filename=filename,
-            default_template_filename=default_template_filename,
-        )
+        self._data: tomlkit.TOMLDocument[str, Any] | None = None
```

Comment:
I'm not totally sure if TOMLDocument is generic like that, you'd have to double check that. Otherwise it's just `tomlkit.TOMLDocument`